File: src/test/java/com/josdem/vetlog/SwaggerConfigTest.java
package com.josdem.vetlog;
import com.josdem.vetlog.configuration.SwaggerConfig;
import io.swagger.v3.oas.models.OpenAPI;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInfo;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SwaggerConfigTest {

    private SwaggerConfig swaggerConfig;
    private static final Logger log = LoggerFactory.getLogger(SwaggerConfigTest.class);

    @BeforeEach
    void setUp() {
        swaggerConfig = new SwaggerConfig();
    }

    @Test
    @DisplayName("Should create OpenAPI object with correct metadata")
    void shouldCreateOpenAPIWithCorrectMetadata(TestInfo testInfo) {
        log.info(testInfo.getDisplayName());

        OpenAPI openAPI = swaggerConfig.vetlogOpenAPI();

        assertEquals("Vetlog API", openAPI.getInfo().getTitle(), "API title should be 'Vetlog API'");
        assertEquals("API documentation for Vetlog backend", openAPI.getInfo().getDescription());
        assertEquals("1.0", openAPI.getInfo().getVersion());
    }
}

File: src/test/java/com/josdem/vetlog/SwaggerConfigTest.kt
/*
  Copyright 2025 Jose Morales contact@josdem.io

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

package com.josdem.vetlog

import com.josdem.vetlog.configuration.SwaggerConfig
import org.junit.jupiter.api.*
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class SwaggerConfigTest {

  private lateinit var swaggerConfig: SwaggerConfig

  @BeforeEach
  fun setUp() {
    swaggerConfig = SwaggerConfig()
  }

  @Test
  @DisplayName("Should create OpenAPI object with correct metadata")
  fun shouldCreateOpenAPIWithCorrectMetadata(testInfo: TestInfo) {
    log.info(testInfo.displayName)

    val openAPI = swaggerConfig.vetlogOpenAPI()

    Assertions.assertEquals("Vetlog API", openAPI.info.title, "API title should be 'Vetlog API'")
    Assertions.assertEquals("API documentation for Vetlog backend", openAPI.info.description)
    Assertions.assertEquals("1.0", openAPI.info.version)
  }

  companion object {
    private val log: Logger = LoggerFactory.getLogger(SwaggerConfigTest::class.java)
  }
}
